airgun/entities/subscription.py

```diff
@@ -0,0 +1,180 @@
+            return
+
+        view.delete_button.click()
+        self._wait_for_process_to_finish(has_manifest=False)
+
+    def read_delete_manifest_message(self):
+        view = self.navigate_to(self, 'Delete Manifest Confirmation')
+        return view.message.read()
+
+    def add(self, entity_name, quantity=1):
+        view = self.navigate_to(self, 'Add')
+        for row in view.table.rows(subscription_name=entity_name):
+            row['Quantity to Allocate'].fill(quantity)
+        view.submit_button.click()
+        self._wait_for_process_to_finish(has_manifest=True)
+
+    def search(self, value):
+        view = self.navigate_to(self, 'All')
+        return view.search(value)
+
+    def provided_products(self, entity_name):
+        view = self.navigate_to(self, 'Details', subscription=entity_name)
+        return [elem.text
+                for elem
+                in view.details.provided_products.browser.elements("./li")]
+
+    def enabled_products(self, entity_name):
+        locator = ("//div[contains(@class, 'list-group')]"
+                   "//div[contains(@class, 'list-group-item-heading')]"
+                   )
+        view = self.navigate_to(self, 'Details', subscription=entity_name)
+        view.enabled_products.select()
+        try:
+            view.browser.wait_for_element(locator, visible=True)
+        except NoSuchElementException:
+            return []
+        return [elem.text for elem in view.browser.elements(locator)]
+
+    def update(self, entity_name, values):
+        # This operation was never implemented in Robottelo (no test
+        # requires it). It is here for consistency, but raises
+        # exception until it is actually needed
+        raise NotImplementedError("Subscriptions update is not implemented")
+
+    def delete(self, entity_name):
+        view = self.navigate_to(self, 'All')
+        for row in view.table.rows(name=entity_name):
+            row['Select all rows'].fill(True)
+        view.delete_button.click()
+        view.confirm_deletion.confirm()
+        self._wait_for_process_to_finish(has_manifest=True)
+
+
+@navigator.register(SubscriptionEntity, 'All')
+class SubscriptionList(NavigateStep):
+    VIEW = SubscriptionListView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Content', 'Subscriptions')
+
+
+@navigator.register(SubscriptionEntity, 'Manage Manifest')
+class ManageManifest(NavigateStep):
+    VIEW = ManageManifestView
+
+    prerequisite = NavigateToSibling('All')
+
+    def step(self, *args, **kwargs):
+        self.parent.manage_manifest_button.click()
+
+    def post_navigate(self, _tries, *args, **kwargs):
+        return self.view.is_displayed
+
+
+@navigator.register(SubscriptionEntity, 'Delete Manifest Confirmation')
+class DeleteManifestConfirmation(NavigateStep):
+    VIEW = DeleteManifestConfirmationView
+
+    prerequisite = NavigateToSibling('Manage Manifest')
+
+    def step(self, *args, **kwargs):
+        wait_for(
```

Comment:
so you need to find out whether element is disabled? if yes - then probably we will need implement wrapper for selenium method: element.is_enabled()

Reply:
I started using Button from widgetastic_patternfly, which already provides `disabled` property. Hopefully it's OK now.